services/api-server/tests/unit/_with_db/conftest.py

```diff
@@ -255,31 +255,39 @@ async def create_fake_api_keys(
     create_product_names: Callable[[PositiveInt], AsyncGenerator[str, None]],
 ) -> AsyncGenerator[Callable[[PositiveInt], AsyncGenerator[ApiKeyInDB, None]], None]:
     async def _generate_fake_api_key(n: PositiveInt):
-        users = create_user_ids(n)
-        products = create_product_names(n)
+        users, products = create_user_ids(n), create_product_names(n)
+        excluded_column = "api_secret"
+        returning_cols = [col for col in api_keys.c if col.name != excluded_column]
+
         for _ in range(n):
             product = await anext(products)
             user = await anext(users)
+            api_key = random_api_key(product, user)
+            plain_api_secret = api_key.pop("api_secret")
             result = await connection.execute(
                 api_keys.insert()
-                .values(**random_api_key(product, user))
-                .returning(sa.literal_column("*"))
+                .values(
+                    api_secret=sa.func.crypt(plain_api_secret, sa.func.gen_salt("bf")),
```

Comment:
I was wondering about this salt.
Does it make sense to have this hard-coded like so? it's not even a constant and it is also in the migration code.
This should never ever be changed correct now? and it will be the same salt in every deployment right?

Reply:
Well, the salt is included in the hash itself... so not required to verify a new value against it.